docs/pubsub-cloudevents.md

```diff
@@ -0,0 +1,374 @@
+# Pub/Sub Transport Binding for CloudEvents - Version 0.3-wip
+
+## Abstract
+
+The Pub/Sub Transport Binding for CloudEvents defines how events are mapped to Pub/Sub
+1.1 request and response messages.
+
+## Status of this document
+
+This document is a working draft.
+
+## Table of Contents
+
+1. [Introduction](#1-introduction)
+
+- 1.1. [Conformance](#11-conformance)
+- 1.2. [Relation to Pub/Sub](#12-relation-to-pubsub)
+- 1.3. [Content Modes](#13-content-modes)
+- 1.4. [Event Formats](#14-event-formats)
+
+2. [Use of CloudEvents Attributes](#2-use-of-cloudevents-attributes)
+
+- 2.1. [datacontenttype Attribute](#21-datacontenttype-attribute)
+- 2.2. [data Attribute](#22-data-attribute)
+
+3. [Pub/Sub Message Mapping](#3-pub/sub-message-mapping)
+
+- 3.1. [Binary Content Mode](#31-binary-content-mode)
+- 3.2. [Structured Content Mode](#32-structured-content-mode)
+- 3.3. [Batched Content Mode](#33-batched-content-mode)
+
+4. [References](#4-references)
+
+## 1. Introduction
+
+[CloudEvents][ce] is a standardized and transport-neutral definition of the
+structure and metadata description of events. This specification defines how the
+elements defined in the CloudEvents specification are to be used in [Pub/Sub][pubsub] requests and response messages.
+
+### 1.1. Conformance
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in [RFC2119][rfc2119].
+
+### 1.2. Relation to Pub/Sub
+
+Events can be produced using a Topic or consumed using Push or Pull based Subscriptions. 
+
+### 1.3. Content Modes
+
+This specification defines three content modes for transferring events:
+_binary_, _structured_ and _batched_. Every compliant implementation SHOULD
+support the _structured_ and _binary_ modes.
+
+In the _binary_ content mode, the value of the event `data` attribute is placed
+into the Pub/Sub payload data as-is, with the `datacontenttype`
+attribute value declaring its media type; all other event attributes are mapped
```

Comment:
Are there limits to the number of attributes that can appear in an event? Pub/Sub does have limitations on the number and size of attributes (https://cloud.google.com/pubsub/quotas#resource_limits).

Reply:
Really good to know! The cloudevents spec has some min-max supported size ideas and they are 64kB.

It would be good to link to this table from this bindings spec.